File: ITapply/ITapply.Models/SearchObjects/CVDocumentSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ITapply.Models.SearchObjects
{
    public class CVDocumentSearchObject : BaseSearchObject
    {
        public int? CandidateId { get; set; }
        public string FileName { get; set; } = string.Empty;
        public bool? IsMain { get; set; }
        public DateTime? UploadDateFrom { get; set; }
        public DateTime? UploadDateTo { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/search_objects/cv_document_search_object.dart
import 'package:itapply_mobile/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'cv_document_search_object.g.dart';

@JsonSerializable()
class CVDocumentSearchObject extends BaseSearchObject {
  int? CandidateId;
  String? FileName;
  bool? IsMain;
  DateTime? UploadDateFrom;
  DateTime? UploadDateTo;

  CVDocumentSearchObject({
    this.CandidateId,
    this.FileName,
    this.IsMain,
    this.UploadDateFrom,
    this.UploadDateTo,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$CVDocumentSearchObjectToJson(this);
}
